DOCS.md

```diff
@@ -171,234 +205,273 @@ The parameters for this method call are:
-        String videoName = (String) data.get("videoName");
-        // [...]
-    };
-    truexAdRenderer.addEventListener(TruexAdRendererConstants.VIDEO_THIRD_QUARTILE, videoThirdQuartileHandler);
+This event will fire anytime a "skip card" is shown to a user as a result of completing a true\[X\] Sponsored Stream interactive ad in an earlier pre-roll.
 
-This event is called whenever a video reaches the third quartile within an engagement.
+This event has no parameters.
 
 
-#### `VIDEO_COMPLETED` *(optional)*
+#### `USER_CANCEL`
 
-    IEventHandler videoCompletedHandler = (Map<String, ?> data) -> {
-        String videoName = (String) data.get("videoName");
-        // [...]
-    };
-    truexAdRenderer.addEventListener(TruexAdRendererConstants.VIDEO_COMPLETED, videoCompletedHandler);
+```java
+IEventHandler userCancelHandler = (Map<String, ?> data) -> {
+    // [...]
+};
+truexAdRenderer.addEventListener(TruexAdRendererConstants.USER_CANCEL, userCancelHandler);
+```
 
-This event is called whenever a video reaches completion within an engagement.
+This event will fire when a user backs out of the true\[X\] interactive ad unit after having opted in. This would be achieved by tapping the "Yes" link to the "Are you sure you want to go back and choose a different ad experience" prompt inside the true\[X\] interactive ad. The user will be subsequently taken back to the Choice Card (with the countdown timer reset to full).
 
+Note that after a `USER_CANCEL`, the user can opt-in and engage with an interactive ad again, so more `OPT_IN` or `OPT_OUT` events may then be fired.
 
-### `TruexAdRenderer` Constants
+This event has no parameters.
 
-These constants will be available in `TruexAdRendererConstants`.
 
+#### `VIDEO_EVENT`
 
-#### `PREROLL`
+```java
+IEventHandler videoEventHandler = (Map<String, ?> data) -> {
+    String type = (String) data.get("type");
+    String videoName = (String) data.get("videoName");
+    String url = (String) data.get("url");
+    // [...]
+};
+truexAdRenderer.addEventListener(TruexAdRendererConstants.VIDEO_EVENT, videoEventHandler);
+```
 
-This value is used to pass into the `init` call, for the `slotType` parameter.
+A `VIDEO_EVENT` is emitted when noteworthy events occur in a video within a true\[X\] unit. These events are differentiated by their `type` value as follows:
 
+* `VIDEO_STARTED`: A video has started
+* `VIDEO_FIRST_QUARTILE`: A video has reached its first quartile
+* `VIDEO_SECOND_QUARTILE`: A video has reached its second quartile
+* `VIDEO_THIRD_QUARTILE`: A video has reached its third quartile
+* `VIDEO_COMPLETED`: A video has reached completion
 
-#### `MIDROLL`
+The parameters for this event are:
 
-This value is used to pass into the `init` call, for the `slotType` parameter.
+* `type`: The type of the emitted video event. Valid values are found in `TruexAdRendererConstants`.
+* `videoName`: The name of the video that emitted the event
+* `url`: The source URL of the video that emitted the event
```

Comment:
Rather than a videoName and a URL, it might make sense to only have a videoName or URL. We use the URL as the videoName for You.i TV and HTML5. I'm not sure if we will be able to get the videoName for Innovid either.